src/main/java/org/jenkins/tools/test/model/PluginRemoting.java

```diff
@@ -106,181 +93,34 @@ private String retrievePomContentFromXmlFile() throws IOException {
-
-        // Generally, when connectionUrl is empty, is implies it is declared in a parent pom
-        // => Only possibility is to deduct github repository from artifactId (crossing fingers it
-        // is not a bizarre repository url...)
-        String oldUrl = transformedConnectionUrl;
-        if (transformedConnectionUrl.isEmpty()) {
-            transformedConnectionUrl =
-                    "scm:git:git://github.com/jenkinsci/"
-                            + pomData.artifactId.replaceAll("jenkins", "")
-                            + "-plugin.git";
-            if (!oldUrl.equals(transformedConnectionUrl)) {
-                LOGGER.log(
-                        Level.WARNING,
-                        "project.scm.connectionUrl is not present in plugin's pom .. isn't it"
-                                + " residing somewhere on a parent pom ?");
-            }
-        }
-
-        // Java.net SVN migration
-        oldUrl = transformedConnectionUrl;
-        transformedConnectionUrl =
-                transformedConnectionUrl.replaceAll(
-                        "(svn|hudson)\\.dev\\.java\\.net/svn/hudson/",
-                        "svn.java.net/svn/hudson~svn/");
-        if (!oldUrl.equals(transformedConnectionUrl)) {
-            LOGGER.log(
-                    Level.WARNING,
-                    "project.scm.connectionUrl is pointing to svn.dev.java.net/svn/hudson/ instead"
-                            + " of svn.java.net/svn/hudson~svn/");
-        }
-
-        // ${project.artifactId}
-        transformedConnectionUrl =
-                transformedConnectionUrl.replaceAll(
-                        "\\$\\{project\\.artifactId\\}", pomData.artifactId);
-
-        // github url like [https://]<username>@github.com/...
-        // => Replaced by git://github.com/...
-        /* Do not change this; it was actually correct if the repo was non-public.
-        oldUrl = transformedConnectionUrl;
-        transformedConnectionUrl = transformedConnectionUrl.replaceAll("(http(s)?://)?[^@:]+@github\\.com", "git://github.com");
-        if(!oldUrl.equals(transformedConnectionUrl)){
-            LOGGER.log(Level.WARNING, "project.scm.connectionUrl is using a github account instead of a read-only url git://github.com/...");
-        }
-        */
-
-        // Convert things like
-        // scm:git:git://git@github.com:jenkinsci/dockerhub-notification-plugin.git
-        oldUrl = transformedConnectionUrl;
-        transformedConnectionUrl =
-                transformedConnectionUrl.replaceAll(
-                        "scm:git:git://git@github\\.com:jenkinsci",
-                        "scm:git:git://github.com/jenkinsci");
-        if (!oldUrl.equals(transformedConnectionUrl)) {
-            LOGGER.log(
-                    Level.WARNING,
-                    "project.scm.connectionUrl should should be accessed in read-only mode (with"
-                            + " git:// protocol)");
-        }
```

Comment:
I have spent months fixing up cases like this and I think most of the fixes have been released. If not then they should be, there is no reason to keep working around this in higher-level code.